signature/cose/envelope_test.go

```diff
@@ -227,73 +220,679 @@ func TestSignErrors(t *testing.T) {
+					//Verify after UnmarshalCBOR
+					env, err := getVerifyCOSE(signingScheme, keyType, size)
+					if err != nil {
+						t.Fatalf("getVerifyCOSE() failed. Error = %s", err)
+					}
+					_, _, err = env.Verify()
+					if err != nil {
+						t.Fatalf("Verify() failed. Error = %s", err)
+					}
+				})
+			}
+
+		}
+	}
+}
+
+func newSignRequest(signingScheme string, keyType signature.KeyType, size int) (*signature.SignRequest, error) {
+	signer, err := getTestSigner(keyType, size)
+	if err != nil {
+		return nil, err
+	}
+	return &signature.SignRequest{
+		Payload: signature.Payload{
+			ContentType: signature.MediaTypePayloadV1,
+			Content:     []byte(payloadString),
+		},
+		Signer:      signer,
+		SigningTime: time.Now(),
+		Expiry:      time.Now().AddDate(0, 0, 1),
+		ExtendedSignedAttributes: []signature.Attribute{
+			{Key: "signedCritKey1", Value: "signedCritValue1", Critical: true},
+			{Key: "signedKey1", Value: "signedValue1", Critical: false},
+		},
+		SigningAgent:  "NotationUnitTest/1.0.0",
+		SigningScheme: signature.SigningScheme(signingScheme),
+	}, nil
+}
+
+func getTestSigner(keyType signature.KeyType, size int) (signature.Signer, error) {
+	switch keyType {
+	case signature.KeyTypeEC:
+		switch size {
+		case 256:
+			leafCertTuple := getECCertTuple(elliptic.P256())
+			certs := []*x509.Certificate{leafCertTuple.Cert, testhelper.GetECRootCertificate().Cert}
+			return signature.NewLocalSigner(certs, leafCertTuple.PrivateKey)
+		case 384:
+			leafCertTuple := getECCertTuple(elliptic.P384())
+			certs := []*x509.Certificate{leafCertTuple.Cert, testhelper.GetECRootCertificate().Cert}
+			return signature.NewLocalSigner(certs, leafCertTuple.PrivateKey)
+		case 521:
+			leafCertTuple := getECCertTuple(elliptic.P521())
+			certs := []*x509.Certificate{leafCertTuple.Cert, testhelper.GetECRootCertificate().Cert}
+			return signature.NewLocalSigner(certs, leafCertTuple.PrivateKey)
+		default:
+			return nil, fmt.Errorf("key size not supported")
+		}
+	case signature.KeyTypeRSA:
+		switch size {
+		case 2048:
+			leafCertTuple := getRSACertTuple(2048)
+			certs := []*x509.Certificate{leafCertTuple.Cert, testhelper.GetRSARootCertificate().Cert}
+			return signature.NewLocalSigner(certs, leafCertTuple.PrivateKey)
+		case 3072:
+			leafCertTuple := getRSACertTuple(3072)
+			certs := []*x509.Certificate{leafCertTuple.Cert, testhelper.GetRSARootCertificate().Cert}
+			return signature.NewLocalSigner(certs, leafCertTuple.PrivateKey)
+		case 4096:
+			leafCertTuple := getRSACertTuple(4096)
+			certs := []*x509.Certificate{leafCertTuple.Cert, testhelper.GetRSARootCertificate().Cert}
+			return signature.NewLocalSigner(certs, leafCertTuple.PrivateKey)
+		default:
+			return nil, fmt.Errorf("key size not supported")
+		}
+	default:
+		return nil, fmt.Errorf("keyType not supported")
+	}
+}
+
+func getRSACertTuple(size int) testhelper.RSACertTuple {
```

Comment:
discussion: which is a better location for the below methods, `testhelper` or this file.

Reply:
nice catch. Moved these two funcs into testhelper.